src/peft/tuners/lora.py

```diff
@@ -301,16 +277,47 @@ def _create_new_module(self, lora_config, adapter_name, target):
 
         return new_module
 
+    def _yield_matching_target_keys(self, model, lora_config: LoraConfig):
+        """For the given model and config, yield all module names that match."""
+
+        # normalize the parameters
+        target_modules = lora_config.target_modules
+        if isinstance(target_modules, str):
+            target_modules = [target_modules]
+        layers_to_transform = lora_config.layers_to_transform
+        if isinstance(layers_to_transform, int):
+            layers_to_transform = [layers_to_transform]
+        layers_pattern = lora_config.layers_pattern
+        if layers_pattern is None:
+            layers_pattern = COMMON_LAYERS_PATTERN
+
+        for key, _ in model.named_modules():
+            if _is_matching_module_name(
+                key,
+                target_modules=target_modules,
+                layers_to_transform=layers_to_transform,
+                layers_pattern=layers_pattern,
+            ):
+                yield key
+
+    def inspect_matching_modules(self, adapter_name: str = "default") -> dict[str, list[str]]:
```

Comment:
The idea was for _users_ to be able to quickly debug if they set up their config correctly. This is just a suggestion, not sure if there is a real world need. If you think there isn't, I would rather remove the function than make it private, because it's just a very thin wrapper around `_yield_matching_target_keys` anyways.

Reply:
I see, great point, I agree with you I think users might be able to use it . In this case it can also live outside the module probably as well no? In any case both solutions are fine for me, feel free to proceed as you wish!

Reply:
> it can also live outside the module probably as well no and make it easily importable

I could move it to a separate function, where would be the best place to put it?

A minor concern of having it as a separate function vs method is that users may try to call it on peft models where it doesn't make sense, whereas the method only lives on those classes where it does make sense.

Reply:
> A minor concern of having it as a separate function vs method is that users may try to call it on peft models where it doesn't make sense, whereas the method only lives on those classes where it does make sense.

Agreed on this ! Makes sense to me, I think it can stay as it is and leave it to super users that manipulates PeftModels